packages/flutter_tools/lib/src/migrations/swift_package_manager_integration_migration.dart

```diff
@@ -557,69 +565,90 @@ $newContent
             .toList()
             .isNotEmpty;
     if (logErrorIfNotMigrated && !migrated) {
-      logger.printError('PBXNativeTarget was not migrated or was migrated incorrectly.');
+      logger.printError('Some PBXNativeTargets were not migrated or were migrated incorrectly.');
     }
     return migrated;
   }
 
-  List<String> _migrateNativeTarget(List<String> lines, ParsedProjectInfo projectInfo) {
-    if (_isNativeTargetMigrated(projectInfo)) {
-      logger.printTrace('PBXNativeTarget already migrated. Skipping...');
-      return lines;
-    }
-
-    final (int startSectionIndex, int endSectionIndex) = _sectionRange('PBXNativeTarget', lines);
-
-    // Find index where Native Target for the Runner target begins.
-    final ParsedNativeTarget? runnerNativeTarget =
-        projectInfo.nativeTargets
-            .where(
-              (ParsedNativeTarget target) => target.identifier == _runnerNativeTargetIdentifier,
-            )
-            .firstOrNull;
-    if (runnerNativeTarget == null) {
-      throw Exception(
-        'Unable to find parsed PBXNativeTarget for ${_xcodeProject.hostAppProjectName} target.',
-      );
-    }
+  void _migrateNativeApplicationTarget(
```

Comment:
I still need to add a check for individual native targets here. Currently the updated migration checks all of them at once, but doesn't yet handle the case if one is migrated but others are not.

Reply:
I currently have this check in place, but I still need to write some tests for it.